src/docs/usr-manual/ch-misc.rst

```diff
@@ -498,6 +498,59 @@ HTML documentation in the ``examples/docs`` directory.
    The examples are designed to mimic real application codes and can serve as
    templates for your own implementations.
 
+Testing the Library
+===================
+
+hypre provides several approaches to test the library, in increasing order of comprehensiveness:
+
+1. **Basic Tests** (Recommended first step):
+   Quick tests to check library functionality (CMake requires ``-DBUILD_TESTING=ON``):
+
+   .. code-block:: bash
+
+      # Single test for each linear system interface
+      make check
+
+      # Test IJ, Struct and SStruct linear solvers in parallel
+      make checkpar
+
+2. **Comprehensive Tests** (CMake only):
+   Test linear solvers for all linear system interfaces (linear-algebraic, Struct and SStruct):
+
+   .. code-block:: bash
+
+      cmake -DBUILD_TESTING=ON ..
+      make -j
+      make test # or ctest
+
+3. **Automated Testing** (Developers only):
+   For thorough testing across different configurations and machines:
+
+   .. code-block:: bash
+
+      cd src/AUTOTEST
+      ./machine_name.sh ../src
```

Comment:
This should be `./test.sh machine_name.sh ../src` .  However, I'm not sure we should be documenting this in the user manual.  There is a `README.txt` in the `AUTOTEST` directory.

Reply:
Great! I added info about AUTOTEST just for completeness in that section. Let me know if you  think it's better to remove it, or just point to the README.txt in the AUTOTEST folder

Reply:
I think we can keep some very minimal info here and point to the README.txt file for details.  That will give us lots of flexibility to make modifications to AUTOTEST without having to touch the online documentation (updating the `README.txt` and `-help` information in the scripts instead).

Reply:
Awesome, thanks!